### ISSUE TITLE
Changed behavior for csv

### ISSUE THREAD

**Issue by JitGub** [drolbr/Overpass-API#371]
I use csv files.

I can see there has been a changed behavior between 2. March 2017 and 10. March 2017.
when search like this:
[out:csv(::"id", "osak:identifier", "osak:street", "osak:street_name", "osak:street_no";true;"\t")];
(
 node["osak:identifier"="0A3F5089124532B8E0440003BA298018"];
 node["osak:identifier"="F09C68B58012484991F3210A0709A04B"];
);
out meta;

I use "http://overpass-api.de/api/interpreter".

Now I get this:
@id	osak:identifier	osak:street	osak:street_name	osak:street_no
341633496	F09C68B58012484991F3210A0709A04B	"Christiansminde, Havekoloni"	"Christiansminde, Havekoloni"	1284
341633516	0A3F5089124532B8E0440003BA298018	Christmas Møllers V.	Christmas Møllers V.	1287

Before I got this:
@id	osak:identifier	osak:street	osak:street_name	osak:street_no
341633496	F09C68B58012484991F3210A0709A04B	Christiansminde, Havekoloni	Christiansminde, Havekoloni	1284
341633516	0A3F5089124532B8E0440003BA298018	Christmas Møllers V.	Christmas Møllers V.	1287

When I use tab as separator I think the " is unnecessary.
My problem is that " is in the 1. line, but not in the 2. line.
Is it a mistake or a permanent change?

**Comment by mmd-osm** [comment]
This change has been introduced in [this commit](https://github.com/drolbr/Overpass-API/commit/bf184de58533017f77692fb01399824463eac503) to implement [rfc 4180](https://tools.ietf.org/html/rfc4180).

It introduces additional double-quotes, in case the string contains a comma (`,`), or a line break: https://github.com/drolbr/Overpass-API/commit/bf184de58533017f77692fb01399824463eac503#diff-cf183a00c814625094cd6da1bcab2609R42 This makes sense for comma-separated files (csv), but in your case, your using a tab character instead.

Maybe this escaping logic should take the actual separator character(s) into consideration as well, rather than assuming it to be comma all the time?

**Comment by JitGub** [comment]
Yes, I think the actual separator character must be taken into consideration.
If I change the separator to .

[out:csv(::"id", "osak:identifier", "osak:street", "osak:street_name", "osak:street_no";true;".")];
(
 node["osak:identifier"="0A3F5089124532B8E0440003BA298018"];
 node["osak:identifier"="F09C68B58012484991F3210A0709A04B"];
);
out meta;

Then I get this:
@id.osak:identifier.osak:street.osak:street_name.osak:street_no
341633496.F09C68B58012484991F3210A0709A04B."Christiansminde, Havekoloni"."Christiansminde, Havekoloni".1284
341633516.0A3F5089124532B8E0440003BA298018.Christmas Møllers V..Christmas Møllers V..1287

Would expect this:
@id.osak:identifier.osak:street.osak:street_name.osak:street_no
341633496.F09C68B58012484991F3210A0709A04B.Christiansminde, Havekoloni.Christiansminde, Havekoloni.1284
341633516.0A3F5089124532B8E0440003BA298018."Christmas Møllers V."."Christmas Møllers V.".1287

**Comment by mmd-osm** [comment]
@JitGub : I deployed a patch on the dev box for testing: http://overpass-turbo.eu/s/nZn

Can you please test if this ok now?

Patch: https://github.com/mmd-osm/Overpass-API/commit/d6628dbda2f63b8dd6b062a0fdde4169b1690730

**Comment by JitGub** [comment]
It is ok now.

I have tried with comma, period, tab and default.
I have tried from a browser and from the command line.
I found no errors.

The only thing I would suggest is to update this page with csv-format:
http://overpass-api.de/output_formats.html

**Comment by drolbr** [comment]
Hi mmd,
this looks like a useful enhancement of RFC 4180. Could you please
- add test cases to src/test-bin/run_unittests_output_csv.sh
- and make a pull request out of the patch
Best regards,
Roland

**Comment by mmd-osm** [comment]
As the patch is fairly small could you please first try a git cherry pick to just includes those 4 lines:

```
git cherry-pick d6628dbda2f63b8dd6b062a0fdde4169b1690730
```

If that doesn't work, I'll create a pull request.

Regarding the test cases, I'm wondering a bit if that means adjusting generate_test_file_meta. Right now, I only see tags like `foo=bar`, `even=yes`, or `some id = some value`. For a good test case, I think it would make sense to include a dot or a comma in some of the key values, and use either dot/comma as a field separator.
